diff --git a/lib/rucio/core/replica.py b/lib/rucio/core/replica.py
index 529cca71af..285d1189e1 100644
--- a/lib/rucio/core/replica.py
+++ b/lib/rucio/core/replica.py
@@ -3021,7 +3021,7 @@ def list_dataset_replicas(scope, name, deep=False, *, session: "Session"):
               with associated metrics and timestamps
     """
 
-    if not deep:
+    if False:   # was `not deep` 
         stmt = select(
             models.CollectionReplica.scope,
             models.CollectionReplica.name,
