From a5b444bb92f2b1b9e49037611b6b96a9e053d3d7 Mon Sep 17 00:00:00 2001
From: Hasan Ozturk <h.ozturk@cern.ch>
Date: Wed, 19 Feb 2025 13:24:25 +0100
Subject: [PATCH] remove rucio commit a37861b due to CMSRucio issue 893

---
 lib/rucio/daemons/reaper/reaper.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/rucio/daemons/reaper/reaper.py b/lib/rucio/daemons/reaper/reaper.py
index ad6dc24924..3c7131aa7e 100644
--- a/lib/rucio/daemons/reaper/reaper.py
+++ b/lib/rucio/daemons/reaper/reaper.py
@@ -16,8 +16,6 @@
 Reaper is a daemon to manage file deletion.
 '''
 
-import concurrent.futures.thread  # noqa (https://github.com/rucio/rucio/issues/6548)
-
 import functools
 import logging
 import random
