diff --git a/lib/rucio/db/sqla/util.py b/lib/rucio/db/sqla/util.py
index fe2fed464..1a5b3f23f 100644
--- a/lib/rucio/db/sqla/util.py
+++ b/lib/rucio/db/sqla/util.py
@@ -313,6 +313,7 @@ def json_implemented(*, session: Optional["Session"] = None) -> bool:
     :type session: Optional[Session]
     :returns: True, if json is supported, False otherwise.
     """
+    return False
     if session is None:
         session = get_session()
