From 71b20fa03fa4d6a23b55b2f16f47c50cf2eb9548 Mon Sep 17 00:00:00 2001
From: Rahul Chauhan <omrahulchauhan@gmail.com>
Date: Fri, 23 Aug 2024 18:45:25 +0200
Subject: [PATCH] Fix dst file report for files that are on tape, SED LYF :(

---
 lib/rucio/transfertool/fts3.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/rucio/transfertool/fts3.py b/lib/rucio/transfertool/fts3.py
index b5eefcde90..81846afc93 100644
--- a/lib/rucio/transfertool/fts3.py
+++ b/lib/rucio/transfertool/fts3.py
@@ -685,7 +685,7 @@ def _is_recoverable_fts_overwrite_error(cls, request: dict[str, Any], reason: st
         dst_type = file_metadata.get('dst_type', None)
         METRICS.counter('overwrite.check.{rsetype}.{rse}').labels(rse=file_metadata["dst_rse"], rsetype=dst_type).inc()

-        if 'Destination file exists and overwrite is not enabled' in (reason or ''):
+        if 'Destination file exists and overwrite is not enabled' in (reason or '') or 'Destination file exists and is on tape' in (reason or ''):
             if cls._dst_file_set_and_file_correct(request, dst_file):
                 if dst_type == 'DISK' or dst_file.get('file_on_tape'):
                     METRICS.counter('overwrite.ok.{rsetype}.{rse}').labels(rse=file_metadata["dst_rse"], rsetype=dst_type).inc()
